core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -2301,12 +2301,18 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
 
   /** The sequence next value function: <code>NEXT VALUE FOR sequence</code>. */
   public static final SqlOperator NEXT_VALUE =
-      new SqlSequenceValueOperator(SqlKind.NEXT_VALUE);
+      new SqlSequenceValueOperator(
+          SqlKind.NEXT_VALUE,
+          ReturnTypes.DOUBLE_NULLABLE,
+          OperandTypes.NILADIC);
```

Comment:
we might have to look at these changes further down the line when complete support for Sequence_generator and sequence value function will be needed. Double_Nullable seems restrictive as of now as a return type. But will merge for now, and ideally would like it to be handled on Mig side unless handled end to end on calcite.